daffodil-core/src/main/scala/org/apache/daffodil/dsom/AnnotatedSchemaComponent.scala

```diff
@@ -186,6 +188,25 @@ trait ResolvesScopedProperties
       }
     }
   }
+
+  val sep = findPropertyOption("separator")
+
+  if(sep.isDefined && (inlineWhiteSpaceDefaulted == InlineWhiteSpace.Disallow) ){
+    val delimSep = sep.toOption.get
+    schemaDefinitionUnless(!delimSep.contains("%LSP;"),
+     """LSP entity cannot act as separator without setting dfdlx:inlineWhiteSpace property to "allow".""")
+    schemaDefinitionUnless(!delimSep.contains("%LSP+;"),
+     """LSP+ entity cannot act as separator without setting dfdlx:inlineWhiteSpace property to "allow".""")
+    schemaDefinitionUnless(!delimSep.contains("%LSP*;"),
+    """LSP* entity cannot act as separator without setting dfdlx:inlineWhiteSpace property to "allow".""")
+    schemaDefinitionUnless(!delimSep.contains("%SP;"),
+     """SP entity cannot act as separator without setting dfdlx:inlineWhiteSpace property to "allow".""")
+    schemaDefinitionUnless(!delimSep.contains("%SP+;"),
+     """SP+ entity cannot act as separator without setting dfdlx:inlineWhiteSpace property to "allow".""")
+    schemaDefinitionUnless(!delimSep.contains("%SP*;"),
+     """SP* entity cannot act as separator without setting dfdlx:inlineWhiteSpace property to "allow".""")
+  }
```

Comment:
I'm not sure if this is the best way to handle this. There are a lot of different DFDL properties aside from the separator that can contain these new character classes (i.e. most properties). So the check probably doesn't want to check for the existence of these special values in specific properties.

Instead, what we probably want to do is pass a boolean into the functions that deal with processing entities that says whether to allow or error when the new entities are found. That way we don't have to check every single property that could have one of these, but instead just need to check where we parse strings.